### PR TITLE
Migrate configuration to use schema fields

### PR DESCRIPTION
**Pull request by shackra**
#### Description

- [ ] Bug fix
- [ ] New feature

#### Useful links:

- [Baton SDK coding guidelines](https://github.com/ConductorOne/baton-sdk/wiki/Coding-Guidelines)
- [New contributor guide](https://github.com/ConductorOne/baton/blob/main/CONTRIBUTING.md)

### CODE REVIEW

File: cmd/baton-fullstory/config.go [hunk 1]
```diff
@@ -1,30 +1,9 @@
...
-	}
-
-	return nil
-}
-
-func cmdFlags(cmd *cobra.Command) {
-	cmd.PersistentFlags().String("api-key", "", "FullStory API Key to authenticate with. ($BATON_API_KEY)")
-}
+var (
+	apikey = field.StringField("api-key", field.WithRequired(true))
```
Comment (mgaeta) [inline cmd/baton-fullstory/config.go:8]: - Add the description `FullStory API Key to authenticate with`.
- Update README
- In `main.go`, add 
 ```go
err := field.Validate(Configuration, v)
	if err != nil {
		return nil, err
	}
```

**Review by mgaeta — CHANGES_REQUESTED** [review]

File: cmd/baton-fullstory/config.go (hunk 1 above)
Comment (shackra) [reply cmd/baton-fullstory/config.go:8]: imagen

No need to add `field.Validate` in main.go tho

Comment (mgaeta) [reply cmd/baton-fullstory/config.go]: Oh cool I hadn't realized `field.Validate()` is automatically called.

**Review by mgaeta — APPROVED** [review]

**Comment by mgaeta** [comment]
This is weird, why is CI failing? Was the integration test succeeding without an API key?

**Pull request merged**